replace SpringContext.getBean() #1062

HuzaifaOmar opened the pull request:
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design
      documentation.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

## Summary
This PR refactors Spring-managed beans by replacing SpringContext.getBean() with constructor-based dependency injection. 
Changes:
- Converted direct calls to SpringContext.getBean() into constructor injection using @Autowired annotation.
- Updated multiple controllers and service classes to use dependency injection.

## Screenshots

## Related Issue
#1062 

## Other

Bahati308 committed:
Detecting unstable selectors

Bahati308 committed:
updated dashboardE2E

Bahati308 committed:
Modified dashboard

Bahati308 committed:
made changes

Bahati308 committed:
Updates to dashboard

Bahati308 committed:
Updated the cytology

Bahati308 committed:
Made updates

Bahati308 committed:
Made updates

Bahati308 committed:
refactored code

Bahati308 committed:
refactored some code

Bahati308 committed:
refactored code

Bahati308 committed:
refactored code

Bahati308 committed:
refactored code

Bahati308 committed:
refactored code

mozzy11 commented on src/main/java/org/openelisglobal/testcalculated/action/util/TestCalculatedUtil.java:
```diff
@@ -35,31 +35,47 @@
 
+    @Autowired
+    public TestCalculatedUtil(TestResultService testResultService, ResultCalculationService resultcalculationService,
+            TestCalculationService calculationService, TestService testService, ResultService resultService,
```
For being consistent with other Classes , let use Field (Property)-Based Dependency Injection as opposed to Constructor-Based Dependency Injection

HuzaifaOmar replied:
Thanks for the feedback @mozzy11 ! I'll update this class to use field-based dependency injection as requested. Just to confirm, do you want this change for only this class, or should I apply it to all the classes I've modified?

mozzy11 reviewed (changes requested):
For being consistent with other Classes , let use Field (Property)-Based Dependency Injection as opposed to Constructor-Based Dependency Injection.

HuzaifaOmar committed:
refactor: Use field injection for TestCalculatedUtil Service

HuzaifaOmar committed:
refactor: Use field injection for TestCalculatedUtil Service

HuzaifaOmar committed:
refactor: Use contructor injection for NonConformingEventsCorrectionActionRestController

HuzaifaOmar committed:
refactor: Use contructor injection for PluginPermissionService

HuzaifaOmar committed:
refactor: Use contructor injection for SampleEntryTestsForTypeProviderRestController

HuzaifaOmar committed:
refactor: Use contructor injection for AllTestsForSampleTypeProviderRestController

HuzaifaOmar committed:
refactor: apply formatter

HuzaifaOmar commented:
Hey @mozzy11, could you please take another look? My PR passed all checks except one—which is failing due to CI using the deprecated actions/cache@v2, not because of my changes. Thanks!

mozzy11 commented:
Thanks @HuzaifaOmar .
Can you use Field (Property)-Based Dependency Injection instead of  Constructor-Based Dependency Injection ??

mozzy11 committed:
Merge branch 'develop' into refactor/replace-springcontext-getbean-1062

HuzaifaOmar commented:
Sorry for the confusion I initially thought you meant only the class you commented on (TestCalculatedUtil). I'll update all classes to use field-based dependency injection.

HuzaifaOmar committed:
Merge branch 'refactor/replace-springcontext-getbean-1062' of github.com:HuzaifaOmar/OpenELIS-Global-2 into refactor/replace-springcontext-getbean-1062

HuzaifaOmar committed:
refactor: Use field injection for NonConformingEventsCorrectionActionRestController

HuzaifaOmar committed:
refactor: Use field injection for PluginPermissionService

HuzaifaOmar committed:
refactor: Use field injection for SampleEntryTestsForTypeProviderRestController

HuzaifaOmar committed:
refactor: Use field injection for AllTestsForSampleTypeProviderRestController

HuzaifaOmar committed:
refactor: apply formatter

HuzaifaOmar commented:
Hey @mozzy11 , I've updated all the classes to use field-based dependency injection as requested. Please take a look and let me know if any further changes are needed. Thanks!

mozzy11 committed:
Merge branch 'develop' into refactor/replace-springcontext-getbean-1062

mozzy11 committed:
Merge branch 'develop' into refactor/replace-springcontext-getbean-1062

mozzy11 committed:
Merge branch 'develop' into refactor/replace-springcontext-getbean-1062

HuzaifaOmar committed:
Merge branch 'develop' into refactor/replace-springcontext-getbean-1062

mozzy11 committed:
Merge branch 'develop' into refactor/replace-springcontext-getbean-1062

mozzy11 commented:
Thanks @HuzaifaOmar .
Can you see the [CI build ](https://github.com/I-TECH-UW/OpenELIS-Global-2/actions/runs/13312724630?pr=1526)failure ??

HuzaifaOmar committed:
Fix: Initialize global variables correctly

The `initializeGlobalVariables` method was being called in the constructor and also using `@PostConstruct`. This commit removes the constructor call to prevent double initialization, ensuring consistent behavior.

HuzaifaOmar commented:
@mozzy11 I've solved the CI Build checks. I think it's ready to merge now

HuzaifaOmar committed:
Merge branch 'develop' into refactor/replace-springcontext-getbean-1062

mozzy11 commented on src/main/java/org/openelisglobal/common/rest/provider/SampleEntryTestsForTypeProviderRestController.java:
```diff
@@ -41,34 +42,34 @@
     }
 
     public SampleEntryTestsForTypeProviderRestController() {
-        initializeGlobalVariables();
```
did you remove this intetionally ??

mozzy11 committed:
Merge branch 'develop' into refactor/replace-springcontext-getbean-1062
